### PR TITLE
Implementing lazy imports for json, toml and yaml modules

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
The imports modules has been moved from usual top lines to methods
where they are actually used.

Signed-off-by: Rahul Tripathi <rahul.tripathi@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Review by selvakumaar5496 — APPROVED** [review]

File: py-utils/src/utils/kv_store/kv_store_collection.py [hunk 1]
```diff
@@ -17,10 +17,7 @@
 
 import configparser
 import errno
-import json
 import os
-import toml
-import yaml
 from json.decoder import JSONDecodeError
```
Comment (sachinpunadikar) [inline py-utils/src/utils/kv_store/kv_store_collection.py:21]: Is this part of json module ? If so shouldn't it be moved to relevant place ?

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
one minor comment. See if it needs any action.

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/kv_store/kv_store_collection.py (hunk 1 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/kv_store/kv_store_collection.py]: just moved import JSONDecodeError just before it is being used.

**Comment by rahul-tripathi-git** [comment]
> one minor comment. See if it needs any action.

Done.

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/kv_store/kv_store_collection.py [hunk 2]
```diff
@@ -79,6 +80,8 @@ def load(self) -> KvPayload:
         return KvPayload(data, self._delim)
 
     def dump(self, data) -> None:
+        """ Saves data onto the file """
+        import yaml
         with open(self._store_path, 'w') as f:
             yaml.dump(data.get_data(), f, default_flow_style=False)
```
Comment (selvakumaar5496) [inline py-utils/src/utils/kv_store/kv_store_collection.py:87]: class TomlKvStore(KvStore):
    """  Represents a TOML File Store """

**Remove one space**
class TomlKvStore(KvStore):
    """ Represents a TOML File Store """

File: py-utils/src/utils/kv_store/kv_store_collection.py [hunk 3]
```diff
@@ -239,6 +246,7 @@ def __init__(self, store_loc, store_path, delim='>'):
 
     def get(self, key):
         """Get pillar data for key."""
```
Comment (selvakumaar5496) [inline py-utils/src/utils/kv_store/kv_store_collection.py:248]: """ Get pillar data for key """

File: py-utils/src/utils/schema/payload.py [hunk 4]
```diff
@@ -81,10 +85,12 @@ def __init__(self, file_path):
...
 
     def _load(self):
+        import yaml
         with open(self._source, 'r') as f:
             return yaml.safe_load(f)
 
     def _dump(self, data):
+        import yaml
         with open(self._source, 'w') as f:
             yaml.dump(data, f)
```
Comment (selvakumaar5496) [inline py-utils/src/utils/schema/payload.py:96]: class Tar(Doc):
    """Represents Tar File"""

**Add spacing**

class Tar(Doc):
    """ Represents Tar File """

**Review by selvakumaar5496 — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/schema/payload.py (hunk 4 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/schema/payload.py:96]: done

File: py-utils/src/utils/kv_store/kv_store_collection.py (hunk 2 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/kv_store/kv_store_collection.py:87]: Done

File: py-utils/src/utils/schema/payload.py (hunk 4 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/schema/payload.py:96]: Done

**Review by selvakumaar5496 — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**